### PR TITLE
Refactor flags to improve unit tests

### PR DESCRIPTION
**Pull request by MarcelMue**
# Changes

This PR refactors the usage of flags for the controller. It now utilizes a config struct which offers multiple benefits, such as easier testing of flag values and more independence between unit tests.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#tests) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#docs) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commits)
- [x] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
Pass a config struct to the reconciler instead of using flags directly.
```

### CODE REVIEW

**Comment by MarcelMue** [comment]
@dibyom Can you check if the direction here is okay? I tried to orient myself a bit towards tektoncd/pipelines - but need to clean up + write new tests still.

File: pkg/reconciler/v1alpha1/eventlistener/controller.go [hunk 1]
```diff
@@ -36,46 +35,49 @@ import (
...
-		configmapLister:     configmapinformer.Get(ctx).Lister(),
-		deploymentLister:    deploymentInformer.Lister(),
-		eventListenerLister: eventListenerInformer.Lister(),
-		serviceLister:       serviceInformer.Lister(),
-		systemNamespace:     os.Getenv("SYSTEM_NAMESPACE"),
-	}
+		reconciler := &Reconciler{
+			KubeClientSet:       kubeclientset,
+			TriggersClientSet:   triggersclientset,
+			configmapLister:     configmapinformer.Get(ctx).Lister(),
```
Comment (dibyom) [inline pkg/reconciler/v1alpha1/eventlistener/controller.go:50]: do we need the configmap lister?

File: pkg/reconciler/v1alpha1/eventlistener/eventlistener.go [hunk 2]
```diff
@@ -111,8 +74,8 @@ type Reconciler struct {
 	eventListenerLister listers.EventListenerLister
 	serviceLister       corev1lister.ServiceLister
 
-	// systemNamespace is the namespace where the reconciler is deployed
-	systemNamespace string
+	// config is the configuration passed in through flags.
```
Comment (dibyom) [inline pkg/reconciler/v1alpha1/eventlistener/eventlistener.go:77]: very minor: config is the configuration options that the Reconciler accepts. 

(We could decide to pass these in via configMaps ore something else in the future)

File: pkg/reconciler/v1alpha1/eventlistener/eventlistener.go [hunk 3]
```diff
@@ -396,7 +359,7 @@ func (r *Reconciler) reconcileDeployment(ctx context.Context, logger *zap.Sugare
 	return nil
 }
 
-func getDeployment(el *v1alpha1.EventListener) *appsv1.Deployment {
+func (r *Reconciler) getDeployment(el *v1alpha1.EventListener) *appsv1.Deployment {
```
Comment (dibyom) [inline pkg/reconciler/v1alpha1/eventlistener/eventlistener.go:362]: If we make this `getDeployment(el *v1alpha1.EventListener, config Config)` the dependency on the `Config` is clearer and the function is a bit easier to test (we don't have to setup an entire reconciler struct)

File: pkg/reconciler/v1alpha1/eventlistener/eventlistener.go [hunk 4]
```diff
@@ -483,7 +446,7 @@ func getDeployment(el *v1alpha1.EventListener) *appsv1.Deployment {
 	}
 }
 
-func getContainer(el *v1alpha1.EventListener) corev1.Container {
+func (r *Reconciler) getContainer(el *v1alpha1.EventListener) corev1.Container {
```
Comment (dibyom) [inline pkg/reconciler/v1alpha1/eventlistener/eventlistener.go:449]: Same as above, I think its a bit clearer to pass in the Config struct as a Param

**Review by dibyom — COMMENTED** [review]
Thanks @MarcelMue This looks much cleaner (and testable) than before.
I have a couple of minor comments and looks like some changes need to be made in test/ to fix the build. Otherwise LGTM!

**Comment by dibyom** [comment]
One other small thing wold be a more descriptive commit message (something like Pass a Config struct to reconciler instead of directly using flags or something.

File: pkg/reconciler/v1alpha1/eventlistener/controller.go (hunk 1 above)
Comment (dibyom) [reply pkg/reconciler/v1alpha1/eventlistener/controller.go:50]: looks like this was there before as well (likely for logging configmap).

File: pkg/reconciler/v1alpha1/eventlistener/eventlistener_test.go [hunk 5]
```diff
@@ -995,12 +1027,29 @@ func TestReconcile(t *testing.T) {
 			Services:       []*corev1.Service{elService},
 			ConfigMaps:     []*corev1.ConfigMap{loggingConfigMap},
 		},
+	}, {
+		name:   "eventlistener with SetSecurityContext false",
```
Comment (MarcelMue) [inline pkg/reconciler/v1alpha1/eventlistener/eventlistener_test.go:1031]: I added this unit test. It is now possible to set a config for each test, ensuring that flags are honored properly.

**Pull request merged**